### PR TITLE
Realtime spelling: Gutter decoration removal, spellchecking without edit

### PR DESCRIPTION
**Pull request by adamconroy**
- Do not show gutter decorations for spell checking errors
- Run spell check on new documents
- Some small optimizations on what touches the spell checker, this is a tiny piece of a larger optimization pass that is coming but made sense to leave in this PR for the linting

Also: The next PR in this series will be off a fresh/rebased branch so the commits don't look like that next time. Apparently known github issue if the target branch is changed even if those commits already exist there.

### CODE REVIEW

File: src/gwt/src/org/rstudio/studio/client/workbench/views/source/editors/text/TextEditingTarget.java [hunk 1]
```diff
@@ -1667,6 +1667,7 @@ public void execute()
       
       syncPublishPath(document.getPath());
       initStatusBar();
+      lintManager_.relintAfterDelay(5000);
```
Comment (jmcphers) [inline src/gwt/src/org/rstudio/studio/client/workbench/views/source/editors/text/TextEditingTarget.java]: Does this change the way non-spellcheck lint behaves? We also probably want to make it this delay configurable via a e.g. a UI pref (no need to add actual UI for the pref at this point)

File: src/gwt/src/org/rstudio/studio/client/workbench/views/source/editors/text/TextEditingTargetSpelling.java [hunk 2]
```diff
@@ -143,7 +143,7 @@ public void onCancelled()
    @Override
    public void invalidateAllWords()
    {
-      lintManager_.forceRelint();
+      lintManager_.relintAfterDelay(-1);
```
Comment (jmcphers) [inline src/gwt/src/org/rstudio/studio/client/workbench/views/source/editors/text/TextEditingTargetSpelling.java]: We should name this special value so it's clear what calling with `-1` does.

File: src/gwt/src/org/rstudio/studio/client/workbench/views/output/lint/LintManager.java [hunk 3]
```diff
@@ -314,9 +315,6 @@ public void onError(ServerError error)
    
    private void showLint(LintContext context, JsArray<LintItem> lint)
    {
-      if (docDisplay_.isPopupVisible() || !docDisplay_.isFocused())
```
Comment (kevinushey) [inline src/gwt/src/org/rstudio/studio/client/workbench/views/output/lint/LintManager.java]: I don't recall exactly why this code was here but it'd be worth confirming that this doesn't interfere with e.g. autocompletion popup menus (e.g. if you were to start typing something, press <TAB> to request completions and the popup menu is shown -- what happens if lint gets shown at the same time?)

File: src/gwt/src/org/rstudio/studio/client/workbench/views/source/editors/text/TextEditingTarget.java (hunk 1 above)
Comment (adamconroy) [reply src/gwt/src/org/rstudio/studio/client/workbench/views/source/editors/text/TextEditingTarget.java]: It'll run the non-spellcheck lint on document load, yeah. In my testing nothing bad has seemed to come up from it. UI pref added.

File: src/gwt/src/org/rstudio/studio/client/workbench/views/source/editors/text/TextEditingTargetSpelling.java (hunk 2 above)
Comment (adamconroy) [reply src/gwt/src/org/rstudio/studio/client/workbench/views/source/editors/text/TextEditingTargetSpelling.java]: 👍

File: src/gwt/src/org/rstudio/studio/client/workbench/views/output/lint/LintManager.java (hunk 3 above)
Comment (adamconroy) [reply src/gwt/src/org/rstudio/studio/client/workbench/views/output/lint/LintManager.java]: Good catch, didn't mean to cart blanch remove that whole line. We do want to run lint on background documents, however, so I've left the focus param out.

**Pull request merged**